### PR TITLE
feat: add sprig to declare default

### PR DESCRIPTION
**Pull request by edouard-lopez**
- feat: add Sprig to templates for additional templating capabilities
- test: add config.Get()
- refactor(template): extract parseJobTemplate to test it

### CODE REVIEW

File: internal/template/parser.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+import (
+	"encoding/base64"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/kubeshop/testkube/internal/config"
+	"github.com/kubeshop/testkube/pkg/log"
+	"github.com/pkg/errors"
```
Comment (vsukhin) [inline internal/template/parser.go:12]: fix import

File: internal/template/parser.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+func LoadConfigFromStringOrFile(inputString, configDir, filename, configType string) (raw string, err error) {
+	var data []byte
+
+	fmt.Printf("TestkubeTemplateJob/inputString: [%v]\n", inputString)
+	fmt.Printf("TestkubeConfigDir: [%v]\n", configDir)
+	fmt.Printf("filename: [%v]\n", filename)
+	fmt.Printf("filepath [%v]\n", filepath.Join(configDir, filename))
+	pwd, _ := os.Getwd()
+	fmt.Printf("pwd: [%v]\n", pwd)
```
Comment (vsukhin) [inline internal/template/parser.go:47]: remove, when finish testing

**Comment by edouard-lopez** [comment]
We introduced a `utils.NewTemplate()` to inject the functions from [`slim-sprig`](https://github.com/go-task/slim-sprig)

```diff
-	t, err := template.New("message").Parse(s.messageTemplate)
+	t, err := utils.NewTemplate("message").Parse(s.messageTemplate)
```

Thanks to @adrien-f for the heavy load :pray: 

@vsukhin As I was exploring, I extracted some methods from _cmd/api-server/main.go_ to [internal/template/parser.go](https://github.com/kubeshop/testkube/pull/4339/files#diff-3fe310a846aa857b279ac5b5b815424015ffe7fc75a1aec2b3e5d152a705bbb5) 
 Could you check that extracting the methods  `LoadConfigFromStringOrFile`, `ParseJobTemplate`, `IsBase64Encoded` has no further impact?

**Comment by vsukhin** [comment]
thank you, guys @edouard-lopez @adrien-f Will check it on Monday morning

**Review by vsukhin — APPROVED** [review]
looks good. @nicufk (Nico developed above methods) ?

**Review by exu — APPROVED** [review]

**Review by nicufk — APPROVED** [review]
LGTM. Nice Job!

**Pull request merged**